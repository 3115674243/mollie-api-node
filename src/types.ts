import createMollieClient from './createMollieClient';
export default createMollieClient;

export * from './createMollieClient';

export { default as List } from './data/list/List';

export { default as Capture } from './data/payments/captures/Capture';
import {
  GetParameters as CapturesGetParameters,
  ListParameters as CapturesListParameters
} from './resources/payments/captures/parameters';
export { CapturesGetParameters, CapturesListParameters };

export { default as Chargeback } from './data/chargebacks/Chargeback';
import {
  ListParameters as ChargebacksListParameters
} from './resources/chargebacks/parameters';
export { ChargebacksListParameters };

export { default as Customer } from './data/customers/Customer';
import {
  CreateParameters as CustomerCreateParams,
  GetParameters as CustomerGetParams,
  ListParameters as CustomersListParams,
  UpdateParameters as CustomerUpdateParams,
  DeleteParameters as CustomerDeleteParams,
} from './resources/customers/parameters';
export { CustomerCreateParams, CustomerGetParams, CustomersListParams, CustomerUpdateParams, CustomerDeleteParams };

export { default as Mandate } from './data/customers/mandates/Mandate';
import { CreateParameters as MandateCreateParams, GetParameters as MandateGetParams, ListParameters as MandatesListParams, RevokeParameters as MandateRevokeParams } from './resources/customers/mandates/parameters';
export { MandateCreateParams, MandateGetParams, MandatesListParams, MandateRevokeParams };

export { default as Method } from './data/methods/Method';
import { GetParameters as MethodGetParams, ListParameters as MethodsListParams } from './resources/methods/parameters';
export { MethodGetParams, MethodsListParams };

export { default as Order } from './data/orders/Order';
import {
  CreateParameters as OrderCreateParams,
  GetParameters as OrderGetParams,
  ListParameters as OrdersListParams,
  UpdateParameters as OrderUpdateParams,
  CancelParameters as OrderCancelParams,
} from './resources/orders/parameters';
export { OrderCreateParams, OrderGetParams, OrdersListParams, OrderUpdateParams, OrderCancelParams };

export { default as OrderLine } from './data/orders/orderlines/OrderLine';
import { UpdateParameters as OrderLineUpdateParams, CancelParameters as OrderLineCancelParams } from './resources/orders/orderlines/parameters';
export { OrderLineUpdateParams, OrderLineCancelParams };

export { default as Payment } from './data/payments/Payment';
import { CreateParameters as PaymentCreateParams, GetParameters as PaymentGetParams, ListParameters as PaymentsListParams, CancelParameters as PaymentCancelParams } from './resources/payments/parameters';
import { CreateParameters as CustomerPaymentCreateParams, ListParameters as CustomerPaymentsListParams } from './resources/customers/payments/parameters';
import { CreateParameters as OrderPaymentCreateParams } from './resources/payments/orders/parameters';
export {
  PaymentCreateParams,
  PaymentGetParams,
  PaymentsListParams,
  PaymentCancelParams,
  CustomerPaymentCreateParams,
  CustomerPaymentsListParams,
  OrderPaymentCreateParams,
};

export { default as Refund } from './data/refunds/Refund';
import { ListParameters as RefundsListParams } from './resources/refunds/parameters';
import {
  CreateParameters as PaymentRefundCreateParams,
  GetParameters as PaymentRefundGetParams,
  ListParameters as PaymentRefundsListParams,
  CancelParameters as PaymentRefundCancelParams,
} from './resources/payments/refunds/parameters';
export { RefundsListParams, PaymentRefundCreateParams, PaymentRefundGetParams, PaymentRefundsListParams, PaymentRefundCancelParams };

export { default as Shipment } from './data/orders/shipments/Shipment';
import { CreateParameters as ShipmentCreateParams, GetParameters as ShipmentGetParams, ListParameters as ShipmentsListParams, UpdateParameters as ShipmentUpdateParams } from './resources/orders/shipments/parameters';
export { ShipmentCreateParams, ShipmentGetParams, ShipmentsListParams, ShipmentUpdateParams };

export { default as Subscription } from './data/subscription/Subscription';
import {
  CreateParameters as SubscriptionCreateParams,
  GetParameters as SubscriptionGetParams,
  ListParameters as SubscriptionsListParams,
  UpdateParameters as SubscriptionUpdateParams,
  CancelParameters as SubscriptionCancelParams,
} from './resources/customers/subscriptions/parameters';
export { SubscriptionCreateParams, SubscriptionGetParams, SubscriptionsListParams, SubscriptionUpdateParams, SubscriptionCancelParams };

export { CardAudience, CardFailureReason, CardLabel, FeeRegion } from './data/global';
export { Issuer } from './data/Issuer';
export { PaymentInclude } from './data/payments/data';
