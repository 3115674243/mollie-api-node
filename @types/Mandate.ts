declare namespace Mollie {
  interface Mandate {
    method: MandateMethod;
    consumerName: string;
    consumerAccount: string;
    consumerBic?: string;
    signatureDate?: string;
    mandateReference?: string;

    // Access token parameters
    testmode?: boolean;
  }

  type MandateDetails = MandateDetailsCreditCard | MandateDetailsDirectDebit;
  interface MandateDetailsDirectDebit {
    consumerName: string;
    consumerAccount: string;
    consumerBic: string;
  }

  interface MandateDetailsCreditCard {
    cardHolder: string;
    cardNumber: string;
    cardLabel: CardLabel;
    cardFingerprint: string;
    cardExpireDate: string;
  }

  type MandateMethod = 'directdebit';
  type MandateStatus = 'valid' | 'pending' | 'invalid';
}
