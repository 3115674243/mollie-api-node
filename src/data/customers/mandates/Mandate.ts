import { MandateData } from './data';
import mandateHelpers from './helpers';
import Seal from '../../../types/Seal';

type Mandate = Seal<MandateData, typeof mandateHelpers>;

export default Mandate;

export function injectPrototypes(input: MandateData): Mandate {
  return Object.assign(Object.create(mandateHelpers), input);
}
