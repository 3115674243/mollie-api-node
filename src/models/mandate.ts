import Model from '../model';

/**
 * The `Mandate` model
 */
export default class Mandate extends Model {
  static STATUS_VALID = 'valid';
  static STATUS_INVALID = 'invalid';
  status: Mollie.MandateStatus;

  constructor(props?: Partial<Mollie.MandateResponse>) {
    super(props);

    const defaults: Mollie.MandateResponse = {
      resource: 'mandate',
      id: null,
      status: null,
      method: null,
      details: null,
      mode: null,
      mandateReference: null,
      signatureDate: null,
      createdAt: null,
      _links: {
        customer: null,
      },
    };

    Object.assign(this, defaults, props);
  }

  /**
   * If the mandate is valid
   * @returns {boolean}
   */
  isValid() {
    return this.status === Mandate.STATUS_VALID;
  }
}
