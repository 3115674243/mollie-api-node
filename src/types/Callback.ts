type Callback<R> = (error: Error | null, result: R) => void;

export default Callback;
