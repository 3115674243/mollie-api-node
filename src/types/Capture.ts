declare namespace Mollie {
  interface ICapture {
    resource: string;
    id: string;
    mode: ApiMode;
    amount: IAmount;
    settlementAmount: IAmount;
    paymentId: string;
    shipmentId?: string;
    settlementId?: string;
    createdAt: string;
    _links: ILinks;
  }

  namespace Params {
    interface Create {
      // Access token parameters
      testmode?: false;
    }
  }
}
