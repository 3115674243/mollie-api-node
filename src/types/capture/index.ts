import { ApiMode, IAmount, ILinks } from '../../types/global';

export interface ICapture {
  resource: string;
  id: string;
  mode: ApiMode;
  amount: IAmount;
  settlementAmount: IAmount;
  paymentId: string;
  shipmentId?: string;
  settlementId?: string;
  createdAt: string;
  _links: ILinks;
}
