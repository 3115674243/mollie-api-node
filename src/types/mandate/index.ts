import { ApiMode, CardLabel, ILinks, IUrl } from '../global';

/**
 * Mandate Response object
 *
 * @see https://docs.mollie.com/reference/v2/mandates-api/get-mandate
 */
export interface IMandate {
  resource: string;
  id: string;
  mode: ApiMode;
  status: MandateStatus;
  method: MandateMethod;
  details: IMandateDetails;
  mandateReference: string;
  signatureDate: string;
  createdAt: string;
  _links: IMandateLinks;

  // Access token parameters
  testmode?: boolean;
}

export interface IMandateLinks extends ILinks {
  customer: IUrl;
}

export type IMandateDetails = IMandateDetailsCreditCard | IMandateDetailsDirectDebit;

export interface IMandateDetailsDirectDebit {
  consumerName: string;
  consumerAccount: string;
  consumerBic: string;
}

export interface IMandateDetailsCreditCard {
  cardHolder: string;
  cardNumber: string;
  cardLabel: CardLabel;
  cardFingerprint: string;
  cardExpireDate: string;
}

export enum MandateMethod {
  directdebit = 'directdebit',
  creditcard = 'creditcard',
}

export enum MandateStatus {
  valid = 'valid',
  invalid = 'invalid',
  pending = 'pending',
}
