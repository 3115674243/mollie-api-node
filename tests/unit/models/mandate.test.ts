import Mandate from '../../../src/models/Mandate';
import { MandateMethod, MandateStatus } from '../../../src/types/mandate';

describe('mandate model', () => {
  it('should instantiate with default values', () => {
    const mandate = new Mandate();

    expect(mandate.isValid()).toBe(false);
    expect(mandate.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const mandateProps = {
      resource: 'mandate',
      id: 'mdt_h3gAaD5zP',
      status: MandateStatus.valid,
      method: MandateMethod.directdebit,
      details: {
        consumerName: 'John Doe',
        consumerAccount: 'NL55INGB0000000000',
        consumerBic: 'INGBNL2A',
      },
      mandateReference: 'YOUR-COMPANY-MD1380',
      signatureDate: '2018-05-07',
      createdAt: '2018-05-07T10:49:08+00:00',
      _links: {
        self: {
          href: 'https://api.mollie.com/v2/customers/cst_4qqhO89gsT/mandates/mdt_h3gAaD5zP',
          type: 'application/hal+json',
        },
        customer: {
          href: 'https://api.mollie.com/v2/customers/cst_4qqhO89gsT',
          type: 'application/hal+json',
        },
        documentation: {
          href: 'https://docs.mollie.com/reference/v2/mandates-api/get-mandate',
          type: 'text/html',
        },
      },
    };
    const mandate = new Mandate(mandateProps);

    expect(mandate.isValid()).toBe(true);
    expect(mandate.toPlainObject()).toMatchSnapshot();
  });
});
