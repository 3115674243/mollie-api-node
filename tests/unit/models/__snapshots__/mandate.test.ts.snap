// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mandate model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "customer": null,
  },
  "createdAt": null,
  "data": undefined,
  "details": null,
  "id": null,
  "isValid": [Function],
  "mandateReference": null,
  "method": null,
  "mode": null,
  "resource": "mandate",
  "signatureDate": null,
  "status": null,
  "toPlainObject": [Function],
}
`;

exports[`mandate model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "customer": Object {
      "href": "https://api.mollie.com/v2/customers/cst_4qqhO89gsT",
      "type": "application/hal+json",
    },
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/mandates-api/get-mandate",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/customers/cst_4qqhO89gsT/mandates/mdt_h3gAaD5zP",
      "type": "application/hal+json",
    },
  },
  "createdAt": "2018-05-07T10:49:08+00:00",
  "data": Object {
    "_links": Object {
      "customer": Object {
        "href": "https://api.mollie.com/v2/customers/cst_4qqhO89gsT",
        "type": "application/hal+json",
      },
      "documentation": Object {
        "href": "https://docs.mollie.com/reference/v2/mandates-api/get-mandate",
        "type": "text/html",
      },
      "self": Object {
        "href": "https://api.mollie.com/v2/customers/cst_4qqhO89gsT/mandates/mdt_h3gAaD5zP",
        "type": "application/hal+json",
      },
    },
    "createdAt": "2018-05-07T10:49:08+00:00",
    "details": Object {
      "consumerAccount": "NL55INGB0000000000",
      "consumerBic": "INGBNL2A",
      "consumerName": "John Doe",
    },
    "id": "mdt_h3gAaD5zP",
    "mandateReference": "YOUR-COMPANY-MD1380",
    "method": "directdebit",
    "resource": "mandate",
    "signatureDate": "2018-05-07",
    "status": "valid",
  },
  "details": Object {
    "consumerAccount": "NL55INGB0000000000",
    "consumerBic": "INGBNL2A",
    "consumerName": "John Doe",
  },
  "id": "mdt_h3gAaD5zP",
  "isValid": [Function],
  "mandateReference": "YOUR-COMPANY-MD1380",
  "method": "directdebit",
  "mode": null,
  "resource": "mandate",
  "signatureDate": "2018-05-07",
  "status": "valid",
  "toPlainObject": [Function],
}
`;
